zerver/lib/actions.py

```diff
@@ -2150,6 +2150,30 @@ def bulk_insert_ums(ums: List[UserMessageLite]) -> None:
         execute_values(cursor.cursor, query, vals)
 
 
+def verify_submessage_sender(
+    *,
+    message_id: int,
+    message_sender_id: int,
+    submessage_sender_id: int,
+) -> None:
+    """Even though our submessage architecture is geared toward
+    collaboration among all message readers, we still enforce
+    the the first person to attach a submessage to the message
+    must be the original sender of the message.
```

Comment:
Ah, I like this way of solving this :grinning: Though I wonder if we may still want to add more comprehensive validation later on? This still allows doing weird things like attaching my own poll to your message in which you posted your own poll. The new poll isn't displayed anywhere, so has no visible impact, but it's still what we probably should consider a corrupt state.

Reply:
My latest PR adds server-side validation for polls.  We'll want the same for todo lists.